edalize/symbiflow.py

```diff
@@ -0,0 +1,176 @@
+
+        nextpnr_edam = {
+                'files'         : self.files,
+                'name'          : self.name,
+                'toplevel'      : self.toplevel,
+                'tool_options'  : {'nextpnr' : {
+                                        'arch' : 'xilinx',
+                                        'nextpnr_as_subtool' : True,
+                                        }
+
+                                }
+                }
+
+        nextpnr = getattr(import_module("edalize.nextpnr"), 'Nextpnr')(nextpnr_edam, self.work_root)
+        nextpnr.configure()
+
+        builddir = self.tool_options.get('builddir', 'build')
+
+        part = self.tool_options.get('part', None)
+        package = self.tool_options.get('package', None)
+
+        assert part is not None, 'Missing required "part" parameter'
+        assert package is not None, 'Missing required "package" parameter'
+
+        partname = part + package
+
+        if 'xc7a' in part:
+            bitstream_device = 'artix7'
+        if 'xc7z' in part:
+            bitstream_device = 'zynq7'
+        if 'xc7k' in part:
+            bitstream_device = 'kintex7'
+
+        makefile_params = {
+                'top' : self.name,
+                'partname' : partname,
+                'bitstream_device' : bitstream_device,
+                'builddir' : builddir,
+            }
+
+        self.render_template('symbiflow-nextpnr-makefile.j2',
+                             'Makefile',
+                             makefile_params)
+
+    def configure_vpr(self):
+        (src_files, incdirs) = self._get_fileset_files(force_slash=True)
+
+        has_vhdl     = 'vhdlSource'      in [x.file_type for x in src_files]
+        has_vhdl2008 = 'vhdlSource-2008' in [x.file_type for x in src_files]
+
+        assert (not has_vhdl and not has_vhdl2008), 'VHDL files are not supported in Yosys'
+        file_list = []
+        timing_constraints = []
+        pins_constraints = []
+        placement_constraints = []
+        user_files = []
+
+        for f in src_files:
+            if f.file_type in ['verilogSource']:
+                file_list.append(f.name)
+            if f.file_type in ['SDC']:
+                timing_constraints.append(f.name)
+            if f.file_type in ['PCF']:
+                pins_constraints.append(f.name)
+            if f.file_type in ['xdc']:
+                placement_constraints.append(f.name)
+            if f.file_type in ['user']:
+                user_files.append(f.name)
+
+        builddir = self.tool_options.get('builddir', 'build')
+
+        # copy user files to builddir
+        for f in user_files:
+            shutil.copy(f, self.work_root + "/" + builddir)
+
+        part = self.tool_options.get('part', None)
+        package = self.tool_options.get('package', None)
+
+        assert part is not None, 'Missing required "part" parameter'
+        assert package is not None, 'Missing required "package" parameter'
```

Comment:
I don't think assertions are the right approach for checking user input. I would use `logger` instead.